Add `&` host/parent filter to `selectTree` path syntax (#121)
Lint-Free-Technology/uix: docs/source/concepts/dom.md

Chains ending with `$` is a special case for convenience, selecting the shadow roots of all elements.

A path may begin with a `&` **host/parent filter** as its first step. It filters the initial element set before any traversal takes place:

- If the current nodes are **ShadowRoot** objects: the filter is tested against the shadow root's **host** element.

> - If the current nodes are regular **Element** objects: the filter is tested against each element's **parentNode**.

Reviewer: ```suggestion
- If the initial element is a regular **Element** the filter is tested against the element's **parentNode**.
```

> Matching is done by directly inspecting element properties — not via CSS selector engine — so it works reliably against shadow-root hosts and parents that are not reachable from the document root. The following tokens are supported (all present tokens must match):

Reviewer: ```suggestion
Matching is done by directly inspecting the parent/host properties — not via CSS selector engine — as required since the host/parent is being filtered. The following tokens are supported (all present tokens must match):
```
